src/prpy/planning/ompl.py

```diff
@@ -175,12 +180,26 @@ def _Plan(self, robot, goal=None, tsrchains=None, timelimit=None,
         # Serialize TSRs into the space-delimited format used by CBiRRT.
         if tsrchains is not None:
             for chain in tsrchains:
-                if chain.constrain or chain.sample_start: 
+                if chain.sample_start:
                     raise UnsupportedPlanningError(
-                        'Only goal tsr is supported by OMPL.')
-                    
-                extraParams += '<{k:s}>{v:s}</{k:s}>'.format(
-                    k='tsr_chain', v=SerializeTSRChain(chain))
+                        'OMPL does not support start TSRs.')
+
+                # Goal TSRs are parsed using the space delimited CBiRRT format.
+                if chain.sample_goal:
+                    extraParams += '<{k:s}>{v:s}</{k:s}>'.format(
+                        k='tsr_chain', v=SerializeTSRChain(chain))
+
+                # Constraint TSRs are parsed by pr_constraint_or_ompl, which
+                # uses a JSON format. This differs from the space delimited
+                # format used by CBiRRT.
+                if chain.constrain:
+                    if self.supports_constraints:
+                        extraParams += '<{k:s}>{v:s}</{k:s}>'.format(
+                            k='tsr_chain_constraint', v=chain.to_json())
```

Comment:
I am fine with adding the JSON format as an option. That change belongs in a separate pull request.

If we do so, I would like to keep `<tsr_chain>` using the CBiRRT format for backwards compatibility.